public/app/features/commandPalette/CommandPalette.tsx

```diff
@@ -146,7 +161,17 @@ const RenderResults = ({ isFetchingSearchResults, searchResults }: RenderResults
   );
 };
 
-const getSearchStyles = (theme: GrafanaTheme2) => {
+const getCommandPalettePosition = (theme: GrafanaTheme2, inputRightPosition: number) => {
+  const paddingSet = parseInt(theme.spacing(2), 10);
+  const screenWidth = window.innerWidth;
+  const lateralSpace = screenWidth - (inputRightPosition + paddingSet);
+  return lateralSpace;
+};
+// eslint-disable-next-line
+const getSearchStyles = (theme: GrafanaTheme2, rightPosition?: any) => {
+  const lateralSpace = getCommandPalettePosition(theme, rightPosition);
```

Comment:
think you can just move all the logic for getting the right position here. something like:

```suggestion
const getCommandPalettePosition = () => {
  const input = document.querySelector('[data-testid="data-testid Command palette trigger"]');
  const inputRightPosition = input?.getBoundingClientRect().right ?? 0;
  const screenWidth = document.body.clientWidth;
  const lateralSpace = screenWidth - inputRightPosition;
  return lateralSpace;
};

const getSearchStyles = (theme: GrafanaTheme2) => {
  const lateralSpace = getCommandPalettePosition();
```

Reply:
also we should use the selector here